tests/core/data_fixtures.py

```diff
@@ -0,0 +1,56 @@
+import anndata as ad
+import mudata as md
+import numpy as np
+import pandas as pd
+import pytest
+import spatialdata as sd
+from scipy.sparse import csr_matrix
+
+
+@pytest.fixture(scope="module")
+def adata():
```

Comment:
Because there is no clean-up needed for simple in-memory objects, we don't need a fixture.

I'd say we should simply use the `core.datasets()` module consistently rather than introducing all these ad hoc datasets.

The good thing then is that we can use the same datasets in our docs.

Reply:
Do we really want to clutter https://docs.lamin.ai/lamindb.core.datasets with even more test datasets? This will only grow...

Reply:
I'll spin this out into an issue where I think more about how we can organize this with our goals:

1. Reuse test datasets in tutorials
2. Reuse fixtures across the many test functions
3. Keep everything sane and readable.

I think this goes beyond the scope of this PR now although I did a tiny tiny step here.

Reply:
> Do we really want to clutter https://docs.lamin.ai/lamindb.core.datasets with even more test datasets? This will only grow...

Yes, I think we should organize all datasets for unit tests and core docs properly in `core.datasets`. The alternative is that they're implicitly scattered all over the place which is much worse. I'm confident we can eliminate many of the datasets that date back years and aren't used anywhere anymore.